ui/hooks/useNftCollectionsMetadata.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useMemo, useState } from 'react';
+import { NameType } from '@metamask/name-controller';
+import { Collection } from '@metamask/assets-controllers';
+import { useSelector } from 'react-redux';
+import { getCurrentChainId } from '../selectors';
+import { fetchNftCollectionsMetadata } from '../store/actions';
+
+export type UseNftCollectionsMetadataRequest = {
+  value: string;
+  type: NameType;
+};
+
+type CollectionsData = {
+  [key: string]: Collection;
+};
+
+export function useNftCollectionsMetadata(
+  requests: UseNftCollectionsMetadataRequest[],
+) {
+  const chainId = useSelector(getCurrentChainId);
+  const [collectionsMetadata, setCollectionsMetadata] =
+    useState<CollectionsData>({});
+
+  const memoisedRequestsWithChainId = useMemo(() => {
```

Comment:
Since chain ID is already a string, can we just memo the contract addresses and pass the arguments as separate dependencies?

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/25692/commits/eed84171c0f7e80457529aa9baec0246fd5e1e00